lib/dialects/snowflake/query.js

```diff
@@ -0,0 +1,338 @@
+'use strict';
+
+const AbstractQuery = require('../abstract/query');
+const sequelizeErrors = require('../../errors');
+const _ = require('lodash');
+const { logger } = require('../../utils/logger');
+
+const ER_DUP_ENTRY = 1062;
+const ER_DEADLOCK = 1213;
+const ER_ROW_IS_REFERENCED = 1451;
+const ER_NO_REFERENCED_ROW = 1452;
+
+const debug = logger.debugContext('sql:snowflake');
+
+class Query extends AbstractQuery {
+  constructor(connection, sequelize, options) {
+    super(connection, sequelize, { showWarnings: false, ...options });
+  }
+
+  static formatBindParameters(sql, values, dialect) {
+    const bindParam = [];
+    const replacementFunc = (_match, key, values_) => {
+      if (values_[key] !== undefined) {
+        bindParam.push(values_[key]);
+        return '?';
+      }
+      return undefined;
+    };
+    sql = AbstractQuery.formatBindParameters(sql, values, dialect, replacementFunc)[0];
+    return [sql, bindParam.length > 0 ? bindParam : undefined];
+  }
+
+  async run(sql, parameters) {
+    this.sql = sql;
+    const { connection, options } = this;
+
+    const showWarnings = this.sequelize.options.showWarnings || options.showWarnings;
+
+    const complete = this._logQuery(sql, debug, parameters);
+
+    if (parameters) {
+      debug('parameters(%j)', parameters);
+    }
+
+    let results;
+
+    try {
+      if (parameters && parameters.length) {
+        results = await new Promise((resolve, reject) => {
+          connection.execute({
+            sqlText: sql,
+            binds: parameters,
+            complete(err, _stmt, rows) {
+              if (err) {
+                console.log(sql);
+                console.error(`Failed to execute statement due to the following error: ${err.message}`);
+                reject(err);
+              } else {
+                console.log(sql);
+                console.log(`Number of rows produced: ${rows.length}`);
```

Comment:
Should this be using the same logging system as everything else?

Reply:
Yes - sorry for the debug left over...will remove it and rely on framework promise processing.